polyfill/lib/casts.mjs

```diff
@@ -44,37 +45,13 @@ export function CastAbsolute(arg, aux) {
 }
 
 export function CastDateTime(arg, aux) {
+  if ('string' === typeof arg) {
+    return DateTime.fromString(arg);
```

Comment:
Note, this code is moved from below the object-handling code in this function; that avoids the intrinsic lookup and also matches the spec. I'm hoping this can land before #252 in any case.

Reply:
Yeah, let's probably make a TODO comment directed to me or sth? To make sure I remember to fix this while rebasing/later on (although a failed test should be good enough, but just in case).